storage/remote/client.go

```diff
@@ -198,11 +212,44 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		err = errors.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
 	if httpResp.StatusCode/100 == 5 {
-		return RecoverableError{err}
+		return RecoverableError{err, defaultBackoff}
+	}
+	if httpResp.StatusCode == 429 {
+		if retryAfterHeader := httpResp.Header.Get("Retry-After"); retryAfterHeader != "" {
+			duration, errRetryAfter := retryAfterDuration(retryAfterHeader)
+			if errRetryAfter != nil {
+				return RecoverableError{errors.Wrap(err, errRetryAfter.Error()), duration}
+			}
+		}
+		return RecoverableError{err, defaultBackoff}
 	}
 	return err
 }
 
+func retryAfterDuration(t string) (durationSecs model.Duration, err error) {
+	var s float64
+	parsedDuration, err := time.Parse(time.RFC1123, t)
+	if err == nil {
+		if parsedDuration.Before(time.Now()) {
+			return invalidRetryAfter, errors.New("retry-after duration cannot be in past")
+		}
+		s = time.Until(parsedDuration).Seconds()
+	} else {
+		// The duration can be simply in seconds.
+		d, err := strconv.Atoi(t)
+		if err != nil {
+			return -1, errors.Wrap(err, "duration must be either in seconds or date-time")
+		}
+		parsedDuration, err := time.ParseDuration(fmt.Sprintf("%ds", d))
+		if err != nil {
+			return -1, errors.Wrap(err, "unable to parse duration")
+		}
```

Comment:
You don't need to do both an `Atoi` and a `ParseDuration`. I would say either do
```
d, err := model.ParseDuration(t+"s")
// handle error
```
or
```
d, err := strconv.Atoi(t)
// handle error
return model.Duration(time.Duration(d) * time.Seconds)
```

Reply:
oh yes, that was unnecessary